diff --git a/pulpcore/content/handler.py b/pulpcore/content/handler.py
index 26a8faa4c..d1b8a6330 100644
--- a/pulpcore/content/handler.py
+++ b/pulpcore/content/handler.py
@@ -613,6 +613,22 @@ class Handler:
             content_disposition = f"attachment;filename={content_artifact.relative_path}"
             parameters = {"ResponseContentDisposition": content_disposition}
             url = artifact_file.storage.url(artifact_file.name, parameters=parameters)
+            # workaround bug in django storages which ignores our parameters
+            from urllib.parse import urlsplit, parse_qs, urlencode
+            from botocore import xform_name
+            splitu = urlsplit(url)
+            qs = parse_qs(splitu.query, keep_blank_values=True)
+            # qs.update(parameters)
+            # We need kebab-case, not CamelCase. Sigh.
+            # So, just merge the lists while transforming the keys
+            for k, v in parameters.items():
+                newk = xform_name(k, '-')
+                if newk in qs.keys():
+                    continue
+                qs[newk] = v
+            url = splitu._replace(query=urlencode(qs, doseq=True)).geturl()
+            del urlsplit, parse_qs, urlencode, xform_name
+            # end workaround
             raise HTTPFound(url)
         elif settings.DEFAULT_FILE_STORAGE == "storages.backends.azure_storage.AzureStorage":
             artifact_file = content_artifact.artifact.file
