diff --git a/pulpcore/content/handler.py b/pulpcore/content/handler.py
index 26a8faa4c..5bd0b1b5a 100644
--- a/pulpcore/content/handler.py
+++ b/pulpcore/content/handler.py
@@ -610,7 +610,7 @@ class Handler:
             return FileResponse(os.path.join(settings.MEDIA_ROOT, filename), headers=headers)
         elif settings.DEFAULT_FILE_STORAGE == "storages.backends.s3boto3.S3Boto3Storage":
             artifact_file = content_artifact.artifact.file
-            content_disposition = f"attachment;filename={content_artifact.relative_path}"
+            content_disposition = f'attachment;filename="{content_artifact.relative_path}"'
             parameters = {"ResponseContentDisposition": content_disposition}
             url = artifact_file.storage.url(artifact_file.name, parameters=parameters)
             raise HTTPFound(url)
