diff --git a/pulpcore/content/handler.py b/pulpcore/content/handler.py
index 26a8faa4c..a9e892aab 100644
--- a/pulpcore/content/handler.py
+++ b/pulpcore/content/handler.py
@@ -7,6 +7,7 @@ from gettext import gettext as _
 from aiohttp.client_exceptions import ClientResponseError
 from aiohttp.web import FileResponse, StreamResponse, HTTPOk
 from aiohttp.web_exceptions import HTTPForbidden, HTTPFound, HTTPNotFound
+from yarl import URL
 
 import django
 
@@ -612,11 +613,13 @@ class Handler:
             artifact_file = content_artifact.artifact.file
             content_disposition = f"attachment;filename={content_artifact.relative_path}"
             parameters = {"ResponseContentDisposition": content_disposition}
-            url = artifact_file.storage.url(artifact_file.name, parameters=parameters)
+            url = URL(artifact_file.storage.url(artifact_file.name, parameters=parameters),
+                      encoded=True)
             raise HTTPFound(url)
         elif settings.DEFAULT_FILE_STORAGE == "storages.backends.azure_storage.AzureStorage":
             artifact_file = content_artifact.artifact.file
-            url = artifact_file.storage.url(artifact_file.name)
+            url = URL(artifact_file.storage.url(artifact_file.name),
+                      encoded=True)
             raise HTTPFound(url)
         else:
             raise NotImplementedError()
